test/e2e/framework/test_context.go

```diff
@@ -356,6 +359,8 @@ func RegisterCommonFlags(flags *flag.FlagSet) {
 	flags.StringVar(&TestContext.NonblockingTaints, "non-blocking-taints", `node-role.kubernetes.io/control-plane`, "Nodes with taints in this comma-delimited list will not block the test framework from starting tests.")
 
 	flags.BoolVar(&TestContext.ListImages, "list-images", false, "If true, will show list of images used for running tests.")
+	flags.BoolVar(&TestContext.listLabels, "list-labels", false, "If true, will show the list of labels that can be used to select tests via -ginkgo.label-filter.")
+	flags.BoolVar(&TestContext.listTests, "list-tests", false, "If true, will show the full names of all tests (aka specs) that can be used to select test via -ginkgo.focus/skip.")
```

Comment:
`-list-tests` and `-list-labels` work as implemented in this commit, but using https://github.com/onsi/ginkgo/issues/1225 would make the output cleaner.

Should I take it out for now or do we want to merge it as-is and improve it later?

I was using `-list-tests` to check for changes in the test texts caused by this PR. There are some because in some cases, I unified the placement of tags and insertion of spaces.

Reply:
Example:
```console
$ go test -v ./test/e2e -args -list-tests
=== RUN   TestE2E
  I0705 10:04:42.204887   72879 e2e.go:117] Starting e2e run "7285667e-32ed-4bca-90cc-e7dfcc2a8fee" on Ginkgo node 1
[1688544282] Kubernetes e2e suite - 4644/7346 specs •••••S•S•S•S••S••••SS••S••S•S•SSS••S••S•••S•••••••S•S•S•SS•SSSS•S•S•S••S•SS••S•S••S•••SSSS•••••••••••S••S•••S••••S••S••S••••••S••SSS••••S••••••••S•S••S•S••••S••••SSSSSS•••••SS•S•••••••••SS••••••SS•S•S••••SS•S••••••••SS••S••••••••S••••S••S•S•S•S••S•S•S••SSS••S•S•SS••••••S••SS•S••SS•SSSS•••••S••S••••••••••••••SSS•S••••SS•S•••••S•SS••S•••S•••••••••S•S•S•SSS•S•••SS•SSS••SS•••S•S•S••SS•••SS•••S•S•S••••S••S••S••S•SS•S••S•••SSSS•S••S•••••SSS•S•S•••••••SSS••••••••S•S•••S•••••S••••••S•S••••••••S•••SS••S•S•S•S••S•S•SS••S•••••S•SSS•S••SS•••••••SS••••••••••S••SSSSSSSS•••••SSS•S•••SSS•S•S•••SS•S•••SS•••••••••S•SS••SSSS•••••S•SSS•••S•SS••••••SS••SS••S•SS•S•••SS•••S••SSS••SS•S•S••SS•S•••S•••••••S•SS••••••••S•S•S•••S•SSS•S••S•••SS•S•SS•••••S•S•S••S••S••••••••••••••S••••••••SSS•S•S•SS•SS•SSSS••SSSS•S•S••S•••SSSSS•••••••S•SS•S•S••S••S•S••S•S••S••••S••S••S•••••SS•S•••S••SS••SS•••••S••SSSS•S•S•S•••••••S••SSS••••SSSS•••••SS•S•••S•••••••S••SS••S•SSSS••S••••••••••S•••S•••••SS•S•SS••S•••S•S•S•••S••SS•S•••••••S••S••S•••••••S•••S••S•S••SSSS•S•S•SS••••SS••SS•••••••••S•••S••S••S••••S••••S•••S•S••SS•••S•SS••SSS••SS••••••S••S•S•••••••••S•S•SS•S•••S••••S•••S•••S••S•SS•••S•••SSS•SSSSS•••S••S••S•••••S••S••••SSSS•••S••••S•••S•S•••SSS•••SS•SS•••S•••SS••••SS•S••SS••••S•••••S••••SSS•••••••••S•SS••SS•SS••••SS••SS••S•S••SSS•••S•••••SS•SSS••SS•S••••••SS•••S•SS•S••SS•••SS•S•••SS••S•••••S••S•SS•SSS•SSSS•S••SS••S•S•••SSSS••S••••S•••S•S•SSSS•••SS•••••••SS•••S•••S•SS••••S•S••S•••S••S••••••••S•••S•SSS•••••••SSSS••S•••••S••SSS••SSS••SS••SSS•S••SSS•••S•••••S•SSS••S•SS••SS•SS•S••SSS•••S•S•SS••SSS•SS•S•S•S•SS•SS•SSSS•••S•S•SS••••SS•••SS•••••••••S•S•S•••••••S••••••••••••••S•••S•••••••••••••S•S••••S••S••S•••S••SS•S•••S••SS••••SS••••SSSS•S••••S••SSS••S•S•SS•••S•••••••S•••••••••SS•••SSS•S•••••SS•S••S•••SS••SS•S•••••••••S••SSS••••S••S••S•SS•••••S••••SS•SSS•••S••S•SS•••••SSS••S••••••••SS••••••SS••••SSS•SSS••S••••SS•S•••••••S•SSSS•S•••••SSSS••SS••••••S•••S•••S•••S•••SS•••SS•••••S•SS••••••S•••S•••••••S•••SS•SSSSS•••SS•S•••••S•SS••SSS•••SSS•SS••S•••S•••••••SSSSS•S•S•S•••••••S•S••••SS•••S••S••••••S•S•••••••••S•S••••S•SS••••SSSS•••••SS••SS•SSSS•S•••S••S••S•••S•••SS••S••••••SS•SS•••S••••SS•••S••SS••S•S••S•••SS••S••S•••••SS•S•••••S••••S•S•••••SS••••••••S•••S••••S•••S••S•••S•S•SS•S•SS•S•S••SS•S••S••S••S••••••SS••SSSS••••••S••••S•••••SS•••••SS••••••S•S•••SS••S•S•S••••S•SSSS•S•S•••••••S•••••S••S•S•S•S••S•••••S••SS•SS•S••••••SS••••SSS••••••SSSSSS•SSS•S•••S••••••S••S•S•S•S••••S•••••••S••SS•••S••S•••••S••SS•••••••S•••S•SSSS•S•SS•••S••S••S•••SS••S••••S••S••••S•••S•••••••S••S•••••SSS••SS•SS••••S•S•••S•••••SS•SSS•••S••S•S•S••SSS•••SSS•SS•S••SS•SS•••••SS•SSSSSSSS•S•SS•S•S•S•S•S••SSS•••S•S••S••S••••S••SS••••••••••••SSS•S•S••S•••S•••S•SS•S•S•S•S••••••SS•••••SSS•S••••••S•••SS••S••SS••••SSS•••S••S•S••••••••••••S•SSS•S•••••S•S••••S•••••••S••SSS•SS••••••••••S••••S•SS••••••••S•••S•S•••SS••SS•S•••S•SSS••SSS•••S•••S•S•SS•S••SS•S•••••••••S•SS••SS••SS•SS•••••••S•••••S•••SS•S•SS•••SSS•••••SSS•SS••••S•••S••SSS••S••••S•••••••••••SSS•••S••S•S•••••••••SSS•SSS•S•••S•S•••SSS•S••••••••S•••••••S•SS••SSS•••••S••••••S••S••SSS••SS•S•••••S•SS•S••SS••S•S•SSS•S•S••SS•••S••••••S•S••SS•••S••••SS••••S•••S••S•••••SS••SS•••S•••••••••••••S•••SS•S•SS•S••••SS•••SS•••••S•S•••••SS••S•S•SS•••S••••SS•S•S•••S••SS•••S•••••••SSS•S••SS••••SS•SSS••S•S•SS••••SS•••S•SS•••S•SS•••••S•S•••••SS•S••S•S•S••S•SSS•••S••S••••S•S••SSS••••SS•S••SS•••S•••S•••SS•SS•S•S•S•S•SS••S•S•••••SS•SSS•S•S••S••SS•••S••••S•S•S••••••SS•••SSS•S••S••SS••••S••••S•••S•••••S•••••S•••S•SS•SSS••S••S••••••S••S•S••S••SS•••S•••S••SSSS•••S•••S•SS•••S••S•S••S•S•••••SSS•S••S••SS••S•SS•SS•••S•S•••••••SS•••S•SS•SS•SS•S•••••••S••S••S••••••S•SS••S•SSS••SSS•S•••••SS••SS••SSS•SSS•••••S••S•SSS•S••••S••SS•••S•SS•••••S•••S•••••••••SS••••S•••SS••SS•S•S•••S••S••S••SS••••••S•SSS•••SS•••••SSSS•S••S•SSSSS•••S••••S••SS••SSS•••SS••S••S••SSS•S•••••••••SS••S•••S•S•••S••S•••S•••SSS•S•S••S••S•SSSS••S•SSS•••S•••S•SS••••••S•S••S•S••••SS•S•••S•S•••••S•••S•SS•SS••••S•S••S•S•S••••S•SSS•••SS••SS•S•••S•••S••••••S•••SS•SSSS•S•SS•S••SS•SS•••••SSSSS••••••S•••S•S••••SSS•••••SS•SS•S•••••SS•SSS••SSS•••SSS•••S•SSS•••SSS••••S••S••S••••••••••SS•SSSSS••••••S••S••••SS•SS•SS•SS•••SS••SS•••S••S•SSSSS•SS•S•••S••SS•SSS•••S•••••SS••S•••S•SSS•S••SS•S••S•••••SS•SS••S•S•S•••SS••••S••••SS•S•S•S••••S•SS••SS••S••S•S•••••••••S••S••••••SSS•S••SSS•S•••S•S••••SSSSSS•••S•SSSSS•••S••••••••S••••SSSS••SSS•SSSS•S•SS•SSSS•SSS••SS•••SS••S•S••••S•••••S••••••S•••SSSSSS••SS••S••••SS•S•SSSSS••SS•••S•••S•SSS•••S••SS•S•S••SS••S••S••••••••S•SS••••SS•S•SS••••SS•S•••S••SS••••SSS•••S•••••••••S••S•S••S•S•••S•S••••••••••SS•••••••S••••S•S••S•S•••S•••••••S•••S•S•S•••••••S••S••••••••S•••••S•SSS•••S•SSS•••••SS••S•SS•S•S•••••••••••••S•S•S•••S•••••SS•SS•S•SS••S••S•S•••SSS•S••••S•S•••SS••S•••••••••SSS••••••SS••••••••••••S••S••S•••S••••••••SS•S••••SS•S••••••••SS•••••••••SS•••••SSS•••••••SSS••S••S•S••••SS•S•••SSS••SS••S•••S•SSS•••SS••S•S•SS•••SS••S•••••••S•••S••••SSS••SS•S••SSSS•SSS•S••••S•SS•••S•••••••S•S••S•••S••S•••S•••S••••SSSS•S••SSS•S•••S•••S••SS•SS••••••••••S••SS••••••••S•S••S•S•SSS••S••••••••SS•••S••S•••S••••S•S••S•S•SSS••••••••••S••S•••S••SS•S••••SS•S•S•SSSS•S•S•••••••••••••••S•••••S•SSSSS••S•••S•SS•S•SSSS••S••S••S•S•••SS••••••••••S•S•S•••S•SSSS••••SSSSS••S•SS•••S•S•S•SSS•SS•SSS•••S••••S•SS••••S•S•SS•S•••S••S•••SS•SS••••SS••S••S•SS•S•SS••••••S••SS••••S•S•S••S•S•••S•••••S•••S•SSSS••SS•SS•S•S••SS•SS••••••SS•••••••SSSS••SS••••S••S••SSS••••S••••SSS••••••S•S•S•••S••S••SS•SS••SS••••••SS••••••S•S••SS•••SS••SSS•••SSS•SS••S•••••S••••••S•••••S••••S••SS••••••S•••SSS•SS•SS•S••SSS•••S••SS•SS•••••••••••••••S••••S•SSS•SS•SS•••S••S•••S•SS•SSSSSS••S•••••S•••S••SS••••••••••S••••S••••S••S•••S••••S•••SSS•••S•••S•S••••S•••••••SSS••••••S••••••S••S••SS•S••••••••SSS•SS•••SS•••••S•S••S••••••••S•S••S•SSS•••SS•••S•S••••S•SS•SS••S•S•S••S•S•SS•S•••S•SSSS••••••S••••S••••S•SS•S••S••••S•SSS••S•S••••••••SS•S•SS•S••SS••••••S•S••••S•••S•••••SS•S••••S•SS•S•S•S•••S•S•S••SSS•••S•S•S••SS•SS•SSSS••S•••••••••••••••••••••••••SSSSS•••S•S•S•S•••••SSS•SS••S•S•S••••SS••S•SS••SSS••S••S•S••SS••SS•••••••S•S••••S••S••S•S••S•••S•••S••S•SSS•S••••••S•••S•••••SS•S••SS••S•S•S••S•SS•S••S••S•S••S•SSS•S••S••SSSSS•••S•SS••••SS•••••••••S••S•S••S••SSSS••••SSSS•••••S••••••••S••S••SS•S•S••SS•SS••SS•••S•S••S••S••S•SSS••S••••••SS••••••••SS•S•SS••SSS•••••S••SS•SS•S•S••••••S•S•S•SSS•S••SS••SS•••••SS•••SS•S•S•••S•S•SS••SSS•SSS•••••••••S•••••S•••S••••S••SS•••SS•••SS•S•S•S••S•SS•••SS••SS•••SS•S•••S••SS••S••••S•SS••••••SS•••S••SS••SSS•••S•••••••SS•••••S••••••S•••S•••••••S•••••••SS•SSSSS••••••S••SS••••SSSS••••S•••••••••••••S•S•S•S•S•••S•••••S•••S•••SSS•S•SSS••••••SS•S•S••SSSS•••SSSS•••••SS•••SS•••••S•S•••S••SS•S•••S••••S•••S•••S••S•S••S••S•S•SS•S•SS•S••••••••SS•S•S•••S•••S••••S•••SS••SS•SS••••••••SS••S•S•SS•SS••SS•SS•S•••••S•SS••••S••••••S••••S••••SS••••SSSS•••S•S•S•SS•SS••••SS••SSS•••SS•••••S••S•••••••SSS•••S••S••••••S•SS•S•SS•S•••S••••S•••SS••••••S•S••••••S••S•S••S••S•SS••••SSS•S•S•SSS•S•SSS•••••••SS•S••S•S•S•S•S•SSSSSS•••••••S•SSS••••S•••••••SS•S•S•S••SSS•••S••S••SSS•••S•S•SSSSSS•••••••S••SS•S•S••••••S•S•S••••••••••SSS•••••••S•S••SS•••••S••S••SS•S•SSSS•S••••S••••SS•SS•SS•••S••••S•••S•SS•••••••SSS••••••••••••S•S•••••S••S•S•S•SSS•••••••••S•SS••SS•••S••SS••SSS••••S••S••••SS••••S••••SSSS•••SSS••S•••••••S•S•SSS••SS•S•S•S•••••••••S•••S•SS•SS•S•S•••••••S•••••SS•SSSS•SSSS•S•••S••SSS•S••••••SS••••••SSSSS•••S

The following spec names can be used with `ginkgo run --focus/skip`:
    [sig-api-machinery] API Streaming (aka. WatchList) [Serial] [Feature:WatchList] should be requested when ENABLE_CLIENT_GO_WATCH_LIST_ALPHA is set
    [sig-api-machinery] API priority and fairness should ensure that requests can be classified by adding FlowSchema and PriorityLevelConfiguration
    [sig-api-machinery] API priority and fairness should ensure that requests can't be drowned out (fairness)
    [sig-api-machinery] API priority and fairness should ensure that requests can't be drowned out (priority)
    [sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] listing mutating webhooks should work [Conformance]
    [sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] listing validating webhooks should work [Conformance]
    [sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] patching/updating a mutating webhook should work [Conformance]
    [sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] patching/updating a validating webhook should work [Conformance]
...
    [sig-windows] Services should be able to create a functioning NodePort service for Windows
    [sig-windows] Windows volume mounts [Feature:Windows] check volume mount permissions container should have readOnly permissions on emptyDir
    [sig-windows] Windows volume mounts [Feature:Windows] check volume mount permissions container should have readOnly permissions on hostMapPath

 SUCCESS! 463.260217ms ESC[38;5;228mYou're using deprecated Ginkgo functionality:ESC[0m
ESC[38;5;228m=============================================ESC[0m
  ESC[38;5;11mImprovements to how reporters emit timeline information means that SuppressProgressReporting is no longer necessary and has been deprecated.ESC[0m

ESC[38;5;243mTo silence deprecations that can be silenced set the following environment variable:ESC[0m
  ESC[38;5;243mACK_GINKGO_DEPRECATIONS=2.9.4ESC[0m

--- PASS: TestE2E (0.70s)
PASS
ok      k8s.io/kubernetes/test/e2e      0.769s
```